src/fluree/db/query/fql/syntax.cljc

```diff
@@ -77,25 +78,32 @@
                              [:string [:fn fn-string?]]
                              [:list [:fn fn-list?]]]
      ::wildcard             [:fn wildcard?]
-     ::var                  [:fn variable?]
-     ::val                  [:fn value?]
-     ::iri                  [:orn
-                             [:keyword keyword?]
-                             [:string string?]]
+     ::var                  [:fn {:decode/json
+                                  (fn [v]
+                                    (log/debug "decoding var:" v)
+                                    (if (string? v)
```

Comment:
Because that fn runs before validation. It even runs on things that malli ultimately decides don't conform to that (e.g. if they're a part of an `:or` schema or similar). So you have to do a little sanity checking and just return the arg if it's not something you want to transform.

For example, the selector (i.e. value of the `:select` key in a query) schema looks like this:

```clojure
[:orn
 [:var ::var]
 [:pred ::iri-pred]
 [:aggregate ::function]
 [:select-map ::select-map]]
```

...which means that this decoder fn will get passed the `{?p [:*]}` map when you have a query like `{:select '{?p [:*]} ...}` because we've told it it could be a `::var`. So it has to know to ignore that.